### PR TITLE
feat(ethPNT): support ethPNT

### PR DESCRIPTION
**Pull request by envin3**
* Update algorand node
* Add check to migrate destination address
* Update Telos Logo
* Add support for PNT swaps using ethPNT token.
ethPNT is an extension of PNT contract; it can be used by switching a toggle after PNT token have been selected:
image

### CODE REVIEW

**Comment by oliviera9** [comment]
What's the motivation behind this PR? What's the need?

**Comment by envin3** [comment]
> What's the motivation behind this PR? What's the need?

It has been requested to support ethPNT directly in the swaps in a similar way to PNT.

File: src/settings/swap-assets.js [hunk 1]
```diff
@@ -1109,7 +1109,7 @@ const swapAssets = [
   },
   {
     address: '0x89ab32156e46f46d02ade3fecbe5fc4243b9aaed',
-    id: 'PNT',
+    id: 'PNT_ON_ETH_MAINNET',
```
Comment (oliviera9) [inline src/settings/swap-assets.js:1112]: Why was this required?
I would be more conservative as this id might be used elsewhere.

File: src/settings/swap-assets.js [hunk 2]
```diff
@@ -1123,6 +1123,26 @@ const swapAssets = [
...
+    name: 'eth-pNetwork',
+    network: 'mainnet',
+    blockchain: 'ETH',
+    decimals: 18,
+    nativeDecimals: 18,
+    withMiniImage: true,
+    isNative: true,
+    nativeSymbol: 'ethPNT',
+    nativeBlockchain: 'ETH',
+    image: 'PNT.svg',
```
Comment (oliviera9) [inline src/settings/swap-assets.js:1139]: why not 
`image: 'ethPNT.svg',` like in migrations?

File: src/hooks/use-assets.js [hunk 3]
```diff
@@ -46,6 +46,19 @@ const useAssetsGroupedByGivenStrategy = (_assets) => {
   }, [_assets])
 }
 
+const useSortPntInflationToken = (_assets) => {
```
Comment (oliviera9) [inline src/hooks/use-assets.js:49]: Where does the sorting occur?

Comment (oliviera9) [reply src/hooks/use-assets.js:49]: In any case, this function is hard to read, difficult to grasp what it does. Please add a comment or consider refactoring it.

**Review by oliviera9 — COMMENTED** [review]
There are some things I didn't understand completely.

File: src/utils/ptokens.js [hunk 4]
```diff
@@ -72,6 +73,28 @@ export const createAsset = async (_asset, _wallets) => {
   return asset
 }
 
+export const createEthPntAsset = async (_asset, _wallets) => {
```
[the remark this answers is not in the document]
Comment (envin3) [reply src/utils/ptokens.js:76]: Good point.

File: src/components/organisms/swapLine/SwapLine.jsx
[the remark this answers is not in the document]
Comment (envin3) [reply src/components/organisms/swapLine/SwapLine.jsx:1]: Yes

[the remark this answers is not in the document]
Comment (oliviera9) [reply src/components/organisms/swapLine/SwapLine.jsx:1]: I am not sure the bridge permits to go from/to ethPNT/PNT interchangeably.

[the remark this answers is not in the document]
Comment (envin3) [reply src/components/organisms/swapLine/SwapLine.jsx:1]: As discussed keeping only PNT on pegout, not ethPNT.

File: src/settings/swap-assets.js (hunk 2 above)
Comment (envin3) [reply src/settings/swap-assets.js:1139]: In the swap lines it has been discussed that given the equivalency of PNT and ethPNT a different logo would be confusing. That logo is to be removed.

File: src/hooks/use-assets.js (hunk 3 above)
Comment (envin3) [reply src/hooks/use-assets.js:49]: It is used here: https://github.com/pnetwork-association/ptokens-dapp/blob/feat/ethPNT-support/src/components/organisms/assetListModal/AssetListModal.jsx#L176
Technically ethPNT has a different address from PNT and has `isNative = true`. Therefore It would be listed by the dapp as an independent token from any other one.
Clearly ethPNT is an extension of PNT so it must be removed from the list somehow. The easiest way found in order to not disrupt every other function which uses `swap-assets.js` list structure is to manipulate the array in the very component that displays the list.

Comment (envin3) [reply src/hooks/use-assets.js:49]: Adding a comment to clarify the function need.

File: src/settings/swap-assets.js (hunk 1 above)
Comment (envin3) [reply src/settings/swap-assets.js:1112]: I need this to match id between migration-assets and swap assets as it is for ethPNT.

File: src/utils/__tests__/swap-valildator.test.js [hunk 5]
```diff
@@ -122,4 +124,11 @@ describe('isValidSwap', () => {
   test('Should return false if from or to assets are not in assets array', () => {
     assets.map((_from) => assets.map((_to) => expect(isValidSwap(_from, _to, migrationAssets)).toBeFalsy()))
   })
+
+  test('Should return true if from asset is ethPNT and to assets are non-native PNT', () => {
```
Comment (oliviera9) [inline src/utils/__tests__/swap-valildator.test.js:128]: Ad a test to check that you cannot go from ethPNT to native PNT

File: src/utils/__tests__/ptokens.test.js
Comment (oliviera9) [inline src/utils/__tests__/ptokens.test.js:1]: Great

File: src/components/organisms/swapLine/SwapLine.jsx [hunk 6]
```diff
@@ -211,22 +224,50 @@ const SwapLine = ({
...
 
+  useEffect(() => {
+    if (asset && asset.id === ETHPNT_ON_ETH_MAINNET) {
+      setSelectEthPNT(true)
+      setShowInfo(true)
+    } else setSelectEthPNT(false)
+  }, [asset, id])
+
+  // This switch provides the ability to use ethPNT or PNT alternatively.
+  // When pressed the selected asset switch between ethPNt or PNT for both pegins and pegouts.
```
Comment (oliviera9) [inline src/components/organisms/swapLine/SwapLine.jsx:255]: revise this comment

File: src/components/organisms/swapLine/SwapLine.jsx [hunk 7]
```diff
@@ -211,22 +224,50 @@ const SwapLine = ({
...
+      if (!address && address !== '') selectFrom(pnt)
+    } else {
+      const ethPnt = assets.find((asset) => asset.id === ETHPNT_ON_ETH_MAINNET)
+      if (!address && address !== '') selectFrom(ethPnt)
+    }
+  }
+
+  const isPNTcase = asset?.id === PNT_ON_ETH_MAINNET || asset?.id === ETHPNT_ON_ETH_MAINNET
+
+  const showPntSwitch = asset && isPNTcase && !migration && !address && address !== ''
```
Comment (oliviera9) [inline src/components/organisms/swapLine/SwapLine.jsx:269]: Maybe I am wrong, but here I don't see where you disable the switch when swapline is used for the destination token, i.e. I expect the switch to appear in the destination swapline when you select PNT. Am I wrong?

Comment (envin3) [reply src/components/organisms/swapLine/SwapLine.jsx:269]: The switch is disabled because of `!address && address !== ''` which is true only for the _from_ swap line component. The _to_ one has address defined.
The same logic is used to show the address input line (https://github.com/pnetwork-association/ptokens-dapp/blob/044089e4244ce50b27f4766147bb829cd17d8cf0/src/components/organisms/swapLine/SwapLine.jsx#L351)

Comment (oliviera9) [reply src/components/organisms/swapLine/SwapLine.jsx:269]: Got it. Maybe add a comment, will we remember this in one year?

Comment (envin3) [reply src/components/organisms/swapLine/SwapLine.jsx:272]: Good point. Added

Comment (oliviera9) [reply src/components/organisms/swapLine/SwapLine.jsx:272]: Thank you

**Review by oliviera9 — APPROVED** [review]
LGTM! good job
